source_handlers_test.go

```diff
@@ -1305,6 +1305,110 @@ func TestSourceEditBadRequest(t *testing.T) {
 	templates.BadRequestTest(t, rec)
 }
 
+func TestSourceEditNameEmptyString(t *testing.T) {
+	newSourceName := ""
+	req := m.SourceEditRequest{
+		Name:               util.StringRef(newSourceName),
+		AvailabilityStatus: util.StringRef("available"),
+	}
+
+	body, _ := json.Marshal(req)
+
+	c, rec := request.CreateTestContext(
+		http.MethodPatch,
+		"/api/sources/v3.1/sources/xxx",
+		bytes.NewReader(body),
+		map[string]interface{}{
+			"tenantID": int64(1),
+		},
+	)
+
+	c.SetParamNames("id")
+	c.SetParamValues("xxx")
+	c.Request().Header.Add("Content-Type", "application/json;charset=utf-8")
```

Comment:
I believe you get here Bad request because you set id = "xxx" and not because new source name is empty string (we parse this id in the handler before the validation function in called)

you can use here any valid source id we have in fixtures, e.g. 1